status/projects.py

```diff
@@ -1049,3 +1049,104 @@ def post(self, project_id):
+                               'days_analysis' : ['all_samples_sequenced', 'best_practice_analysis_completed'],
+                               'days_data_delivery' : ['best_practice_analysis_completed', 'all_raw_data_delivered'],
+                               'days_close' : ['all_raw_data_delivered', 'close_date']
+                             }
+
+        date_fields = list(flatten(def_dates_ongoing.values()))
+        statuses = ['ongoing', 'reception control']
+        view_calls = []
+
+        view = self.application.projects_db.view("project/summary_status", descending=True)
+        for status in statuses:
+            view_calls.append(view[[status, 'Z']:[status, '']])
+        for row in itertools.chain.from_iterable(view_calls):
+            proj_id_name_lib = row.value['project_name']+' ('+row.key[1]+')'+'| '+row.value['details']['library_construction_method']
+            proj_val = row.value
+            for date_type, date in proj_val['summary_dates'].items():
+                proj_val[date_type] = date
+            if row.key[0] == 'ongoing':
+                for k, v in proj_val['project_summary'].items():
+                    proj_val[k] = v
+
+            is_fin_lib = False
+            if 'by user' in proj_val.get('library_construction_method', '-').lower():
+                is_fin_lib = True
+
+            for key, value in def_dates_rec_ctrl.items():
+                start_date = value[0]
+                end_date = value[1]
+                date_val = self._calculate_days_in_status(proj_val.get(start_date),
+                                                               proj_val.get(end_date))
+                projects[proj_id_name_lib] = { key: date_val }
+
+            if row.key[0] == 'ongoing':
+                for key, value in def_dates_ongoing.items():
+                    if key == 'days_seq_start':
+                        if is_fin_lib:
+                            start_date = value[0][1]
+                        else:
+                            start_date = value[0][0]
+                    else:
+                        start_date = value[0]
+                        end_date = value[1]
+                    if key in ['days_prep', 'days_prep_start'] and is_fin_lib:
+                        date_val = 0
+                    else:
+                        date_val = self._calculate_days_in_status(proj_val.get(start_date),
+                                                                       proj_val.get(end_date))
+
+                    projects[proj_id_name_lib][key] = date_val
+
+        #Delete statuses with 0 days
+        for k,v in projects.items():
+            min_days = 0
+            for k2 in list(v):
+                if v[k2] <= min_days:
+                    del projects[k][k2]
+ 
+        #Get list of projects with status and days containing only last status
+        t_data = [(k,k2,v2) for k,v in projects.items() for k2,v2 in v.items() if k2 in list(v.items())[-1]]
```

Comment:
the variable `v` here seems to be defined (on the right half of it at least) from the for loop above, is that intentional?

Reply:
Not sure I understand 🤔  do you mean that I should add the code to the loop above?